internal/signaling/peer.go

```diff
@@ -90,11 +90,14 @@ func (p *Peer) RequestConnection(ctx context.Context, otherID string) error {
 }
 
 func (p *Peer) ForwardMessage(ctx context.Context, raw []byte) {
-	logger := logging.GetLogger(ctx)
-	err := p.conn.Write(ctx, websocket.MessageText, raw)
-	if err != nil && !util.IsPipeError(err) {
-		logger.Warn("failed to forward message", zap.Error(err))
-	}
+	// Don't block our receive loop
+	go func() {
```

Comment:
I would move this responsibility to the caller. This ForwardMessage is passed to Subscribe, in there it should just call this function as a goroutine.

Reply:
If we ever have a fast callback in the future, calling it in a new goroutine would just slow it down. That's why I thought leaving this up to the callee would be better as it would allow us to decide per callback.

Reply:
I think in this case, i rather have the small performance impact of code that doesn't exist yet and keep the store API easier to use/read. :) But I'm also fine with this. 👍🏻